extensions/oidc-client/runtime/src/main/java/io/quarkus/oidc/client/runtime/OidcClientRecorder.java

```diff
@@ -200,7 +171,10 @@ private static void setGrantClientParams(OidcClientConfig oidcConfig, MultiMap g
                 LOG.tracef("Discovery has failed, status code: %d", resp.statusCode());
                 return null;
             }
-        });
+        }).onFailure()
```

Comment:
@Yelzhasdev what happens if it is an HTTP-level response error, for example, 401 ? This block must only run if it is a connection error

Reply:
fixed

Reply:
@Yelzhasdev, nice - can you confirm please this is what vert.x  web client throws in case of the connection failure ? I was referring to `ConnectionException` as a general connection exception... vert.x might wrap it, etc...

Reply:
Yes, for example for connection timeout there is an exception io.netty.channel.ConnectTimeoutException, which is a child of java.net.ConnectException